src/Services/NumberFormatter.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: vic
+ * Date: 18.12.7
+ * Time: 23.17
+ */
+
+namespace App\Services;
+
+
+class NumberFormatter implements NumberFormaterInterface
+{
+    /**
+     * @param float $number
+     * @return string|null
+     */
+    public function formatNumber(float $number): ?string
+    {
+        if ($number >= 999500 || $number <= -999500){
+            $formatedNumber = $this->formatToMillions($number);
+            return $formatedNumber;
+        }elseif ($number >= 99950 && $number < 999500 || $number <= -99950 && $number > -999500){
+            $formatedNumber = $this->formatToThousands($number);
+            return $formatedNumber;
+        }elseif ($number >= 1000 && $number < 99950 || $number <= -1000 && $number > -99950) {
+            $formatedNumber = $this->formatToIntegerWithGap($number);
+            return $formatedNumber;
+        }elseif ($number >= 0 && $number < 1000 || $number < 0 && $number > -1000){
+            $formatedNumber = $this->formatToDec($number);
+            return $formatedNumber;
+        }
+        else{
+            return null;
+        }
+    }
+
+    /**
+     * @param float $number
+     * @return string
+     */
+    public function formatToMillions (float $number): string
```

Comment:
Yra gera praktika visas nebūtinas funkcijas daryti `private`.

Nes jei naudoji `public` – tai reiškia visi kiti irgi galės naudotis, ko pasekoje tau reikia parašyti testus ir kiekvienam iš šių metodų.

Bet pas tave yra testai tik vienam metodui:
https://github.com/aurelijusb/kickstart/blob/8af2cceafacd605e67022a4f945374ea68300114/tests/Services/NumberFormatterTest.php#L33

Plačiau:
* https://www.quora.com/Should-you-unit-test-private-methods-on-a-class